src/lib.rs

```diff
@@ -154,23 +155,30 @@ impl RandomXDataset {
                 dataset_count: count,
             };
             unsafe {
+                //no way to check if this fails, c code does not return anything
                 randomx_init_dataset(result.dataset, cache.cache, start, count);
             }
             Ok(result)
         }
     }
 
-    pub fn count(&self) -> c_ulong {
-        unsafe { randomx_dataset_item_count() }
+    pub fn count(&self) -> Result<u64, CreationError> {
+        match unsafe { randomx_dataset_item_count() } {
+            0 => Err(CreationError),
```

Comment:
I don't think the return of zero is necessarily an error here.

Reply:
I might not be an error. But its very likely that the dataset item count will never be 0. 
0 is the default u64 value. Thats why I am taking a guess that if this is 0, it failed and the u64 has its default value. 

Do you think this is perhaps too dangerous?

Reply:
I am inclined to agree with you that this in all probability should not be zero.